### PR TITLE
Added Shell Short [Java]

### PR DESCRIPTION
**Pull request by jsroyal**
Fixes #392 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [X] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [X] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [X] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [X] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [X] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: shell_sort/ShellSort.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Worst case time complexity = O(n^2)
+// Best case complexity = O(nlog(n))
+// n is input size
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: Please add all the complexity related comment to the related function. Also use blocks for more than one comment.

File: shell_sort/ShellSort.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                        data[k] = data[k + i];
+                        data[k + i] = temp;
+                    }
+                }
+            }
+        }
+        return data;
+    }
+    // print function
+    public static void print(int[]  data) {
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: Add a blank line before 23.

File: shell_sort/ShellSort.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*Shell Short function 
+Worst case time complexity = O(n^2)
+ Best case complexity = O(nlog(n))
+n is input size*/
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: @jsroyal  the comment block is related to `shellSort()`.
```
/*
Shell Short function 
Worst case time complexity = O(n^2)
Best case complexity = O(nlog(n))
n is input size
*/
```
this block belongs to line 6. Also look at the  @[Javadoc](https://google.github.io/styleguide/javaguide.html#s7-javadoc) in Google Java Style Guide.

Comment (jsroyal) [inline shell_sort/ShellSort.java]: @mohitkyadav Is that now ok?

File: shell_sort/ShellSort.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+ 
+
+public class ShellSort {
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: Remove 2 blank lines. Remove trailing space at the end of line 3.

File: shell_sort/ShellSort.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
+ 
+
+public class ShellSort { 
+/**
+ * sorting function
+ * Worst case time complexity = O(n^2)
+ * Best case complexity = O(nlog(n))
+ * n is input size
+ */
+    public static int[] shellSort(int[] data) {
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: Please indent the comment block with the `shellSort()`.

File: shell_sort/ShellSort.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                        int temp = data[k];
+                        data[k] = data[k + i];
+                        data[k + i] = temp;
+                    }
+                }
+            }
+        }
+        return data;
+    }
+
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: Remove the trailing white space.

File: shell_sort/ShellSort.java (hunk 3 above)
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: @jsroyal use 4-space instead of a tab.

File: shell_sort/ShellSort.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+public class ShellSort {
+	/**
+	* sorting function
+	* Worst case time complexity = O(n^2)
+	* Best case complexity = O(nlog(n))
+	* n is input size
+	*/
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: `invalid indent style: tab, expected: space`
Use 4 space instead of a tab.

File: shell_sort/ShellSort.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+                        int temp = data[k];
+                        data[k] = data[k + i];
+                        data[k + i] = temp;
+                    }
+                }
+            }
+        }
+        return data;
+    }
+
```
Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: Please remove space from this line.

File: shell_sort/ShellSort.java (hunk 7 above)
Comment (jsroyal) [inline shell_sort/ShellSort.java]: >Use 4 space 

How many is there?

Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: There's a `Tab`.
try selecting this space inside the double quotes
TAB => "	"
SPACE => "    "

Comment (jsroyal) [inline shell_sort/ShellSort.java]: Why do you prefer space over tab??

Comment (mohitkyadav) [inline shell_sort/ShellSort.java]: As mentioned in the `.editorconfig`, space indentation should be used for java files.

**Pull request merged**